### PR TITLE
Deprecate iotcentral command group

### PR DESCRIPTION
**Pull request by r-delgadillo**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [x] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [x] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [x] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.

### CODE REVIEW

**Comment by r-delgadillo** [comment]
Command group 'az iotcentral' has been deprecated on the Azure CLI. Deprecating the same command group in extension.

PR for Azure CLI
https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/HISTORY.rst

File: azext_iot/commands.py [hunk 1]
```diff
@@ -164,11 +164,16 @@ def load_command_table(self, _):
         cmd_group.command("show", "iot_dps_registration_get")
         cmd_group.command("delete", "iot_dps_registration_delete")
 
+    with self.command_group(
+        'iotcentral', command_type=iotcentral_ops,
+        deprecate_info=self.deprecate(redirect='iot central', hide=True)
+    ) as cmd_group:
+        pass
+
     with self.command_group("iotcentral app", command_type=iotcentral_ops) as cmd_group:
```
[the remark this answers is not in the document]
Comment (r-delgadillo) [reply azext_iot/commands.py:173]: We have to leave the command there for at least 3 months before removing it.

[the remark this answers is not in the document]
Comment (r-delgadillo) [reply azext_iot/commands.py:173]: The changes in this PR will mark anything under `az iotcentral` as deprecated

[the remark this answers is not in the document]
Comment (digimaun) [reply azext_iot/commands.py:173]: If possible we should. Also the deprecated `iotcentral device-twin` command group. @r-delgadillo is there a reason they are still around?

[the remark this answers is not in the document]
Comment (r-delgadillo) [reply azext_iot/commands.py:173]: We have a work item in our backlog to completely  remove all commands under `az iotcentral` in June or July, I forgot which month specifically. We'd like to keep these commands available for customers who are using them to give them sufficient time for them to roll off these commands and start using the new command group. Hence that's the reason we are hiding the existing command group so no other customers become aware of those commands.

**Pull request merged**